shtdlib.sh

```diff
@@ -36,7 +36,7 @@ function whichs {
 # Set strict mode only for non-interactive (see bash tab completion bug):
 # https://github.com/scop/bash-completion/issues/44
 # https://bugzilla.redhat.com/show_bug.cgi?id=1055784
-if ! ${interactive} ; then
+if ${strict_mode:-true} && ! ${interactive} ; then
```

Comment:
Should we make a comment block at the top of the file listing possible variables? E.g. `strict_mode`, `verbosity`, etc.

Reply:
I'm not sure that's the right way to document these.

In my opinion documentation/comments should preferably accompany the code it/they refer to in case the code is updated or needed for reference/context.

If we add a "readme" section to the top then I'm afraid it will get out of sync and I wonder how many users would start reading the header of the file for information about specific functionality or functions.

If more generic non function specific documentation is needed I think we might be better off adding something high level in the overall project README?

Reply:
Project README would have the same effect. 
 
I'll approve and leave it to you

Reply:
Added notes to the readme for the other MR since it touches both.